File: Team C (Season 2)/Week 2/Greed/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    queue = [(i,q) for i, q in enumerate(priorities)]
    while True :
        cur = queue.pop(0)
        if any(cur[1] < q[1] for q in queue) :
            queue.append(cur)
        else :
            answer += 1
            if cur[0] == location :
                return answer

File: Team C (Season 2)/Week 2/Cliff/프로세스.swift
import Foundation

func dequeue(_ array: inout [Int]) -> Int? {
    guard !array.isEmpty else {
        return nil
    }

    let first = array[0]
    array = Array(array.dropFirst())

    return first
}

func process(_ priorities: [Int], _ location: Int) -> Int {
    var queue = priorities
    var location = location
    
    var answer = 0
    
    while !queue.isEmpty {
        let max = queue.max()!
        guard let first = dequeue(&queue) else {
            return 0
        }
        location -= 1
        
        if first != max {
            queue.append(first)
            if location < 0 {
                location = queue.count - 1 // 로케이션 리셋
            }
        } else {
            answer += 1
            if location < 0 {
                break
            }
        }
    }
    
    return answer
}

process([2, 1, 3, 2], 2)
process([1, 1, 9, 1, 1, 1], 0)